src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -76,12 +75,12 @@ public boolean hitExecutionNeeded(SearchContext context) {
     public void hitExecute(SearchContext context, HitContext hitContext) {
         Map<String, HighlightField> highlightFields = newHashMap();
         for (SearchContextHighlight.Field field : context.highlight().fields()) {
-            List<String> fieldNamesToHighlight;
+            Set<String> fieldNamesToHighlight;
```

Comment:
how can the same field be highlighted twice if we use a set that doesn't allow for duplicates? not sure what I'm missing here. Anyway, agree that the problem is probably in mappings code, but I am not familiar with it, somebody else will have to jump in on that.